### PR TITLE
#16293 - Add `ext/pcre/php_pcre.h` header for PHP8.2

### LINKED ISSUE #16293: [BUG]: Unable to install using pecl on PHP 8.2
**Describe the bug**

Unable to install using pecl on PHP 8.2

**To Reproduce**

```
docker run -it --rm php:8.2 bash
pecl install phalcon-5.2.0
...
make: *** [Makefile:210: phalcon.lo] Error 1
ERROR: `make' failed
```

**Expected behavior**

Phalcon must be able to be installed using pecl.

**Screenshots**

<details>
<summary>Part of the build log</summary>
<pre>
<code>
downloading phalcon-5.2.0.tgz ...
Starting to download phalcon-5.2.0.tgz (954,435 bytes)
...............................................................................................................................................................
..............................done: 954,435 bytes
5 source files, building
running: phpize
Configuring for:
PHP Api Version:         20220829
Zend Module Api No:      20220829
Zend Extension Api No:   420220829
building in /tmp/pear/temp/pear-build-defaultuserNSBfOD/phalcon-5.2.0
running: /tmp/pear/temp/phalcon/configure --with-php-config=/usr/local/bin/php-config
checking for grep that handles long lines and -e... /bin/grep
checking for egrep... /bin/grep -E
checking for a sed that does not truncate output... /bin/sed
checking for pkg-config... /usr/bin/pkg-config
checking pkg-config is at least version 0.9.0... yes
checking for cc... cc
checking whether the C compiler works... yes
checking for C compiler default output file name... a.out
checking for suffix of executables...
checking whether we are cross compiling... no
checking for suffix of object files... o
checking whether we are using the GNU C compiler... yes
checking whether cc accepts -g... yes
checking for cc option to accept ISO C89... none needed
checking how to run the C preprocessor... cc -E
checking for icc... no
checking for suncc... no
checking for system library directory... lib
checking if compiler supports -Wl,-rpath,... yes
checking build system type... x86_64-pc-linux-gnu
checking host system type... x86_64-pc-linux-gnu
checking target system type... x86_64-pc-linux-gnu
checking for PHP prefix... /usr/local
checking for PHP includes... -I/usr/local/include/php -I/usr/local/include/php/main -I/usr/local/include/php/TSRM -I/usr/local/include/php/Zend -I/usr/local/in
clude/php/ext -I/usr/local/include/php/ext/date/lib
checking for PHP extension directory... /usr/local/lib/php/extensions/no-debug-non-zts-20220829
checking for PHP installed headers prefix... /usr/local/include/php
checking if debug is enabled... no
checking if zts is enabled... no
checking for gawk... no
checking for nawk... nawk
checking if nawk is broken... no
checking whether to enable phalcon... yes, shared
checking PHP version... checking whether HAVE_BUNDLED_PCRE is declared... yes
checking for ext/pcre/php_pcre.h... yes
checking whether HAVE_JSON is declared... yes
checking for ext/json/php_json.h... yes
checking for a sed that does not truncate output... /bin/sed
checking for ld used by cc... /usr/bin/ld
checking if the linker (/usr/bin/ld) is GNU ld... yes
checking for /usr/bin/ld option to reload object files... -r
checking for BSD-compatible nm... /usr/bin/nm -B
checking whether ln -s works... yes
checking how to recognize dependent libraries... pass_all
checking for ANSI C header files... yes
checking for sys/types.h... yes
checking for sys/stat.h... yes
checking for stdlib.h... yes
checking for string.h... yes
checking for memory.h... yes
checking for strings.h... yes
checking for inttypes.h... yes
checking for stdint.h... yes
checking for unistd.h... yes
checking dlfcn.h usability... yes
checking dlfcn.h presence... yes
checking for dlfcn.h... yes
checking the maximum length of command line arguments... 1572864
checking command to parse /usr/bin/nm -B output from cc object... ok
checking for objdir... .libs
checking for ar... ar
checking for ranlib... ranlib
checking for strip... strip
checking if cc supports -fno-rtti -fno-exceptions... no
checking for cc option to produce PIC... -fPIC
checking if cc PIC flag -fPIC works... yes
checking if cc static flag -static works... yes
checking if cc supports -c -o file.o... yes
checking whether the cc linker (/usr/bin/ld -m elf_x86_64) supports shared libraries... yes
checking whether -lc should be explicitly linked in... no
checking dynamic linker characteristics... GNU/Linux ld.so
checking how to hardcode library paths into programs... immediate
checking whether stripping libraries is possible... yes
checking if libtool supports shared libraries... yes
checking whether to build shared libraries... yes
checking whether to build static libraries... no

creating libtool
appending configuration tag "CXX" to libtool
configure: patching config.h.in
configure: creating ./config.status
config.status: creating config.h
running: make
/bin/bash /tmp/pear/temp/pear-build-defaultuserNSBfOD/phalcon-5.2.0/libtool --mode=compile cc -I. -I/tmp/pear/temp/phalcon -I/tmp/pear/temp/pear-build-defaultu
serNSBfOD/phalcon-5.2.0/include -I/tmp/pear/temp/pear-build-defaultuserNSBfOD/phalcon-5.2.0/main -I/tmp/pear/temp/phalcon -I/usr/local/include/php -I/usr/local
/include/php/main -I/usr/local/include/php/TSRM -I/usr/local/include/php/Zend -I/usr/local/include/php/ext -I/usr/local/include/php/ext/date/lib  -DHAVE_CONFIG
_H  -g -O2 -D_GNU_SOURCE    -DZEND_COMPILE_DL_EXT=1 -c /tmp/pear/temp/phalcon/phalcon.zep.c -o phalcon.lo  -MMD -MF phalcon.dep -MT phalcon.lo
mkdir .libs
 cc -I. -I/tmp/pear/temp/phalcon -I/tmp/pear/temp/pear-build-defaultuserNSBfOD/phalcon-5.2.0/include -I/tmp/pear/temp/pear-build-defaultuserNSBfOD/phalcon-5.2.
0/main -I/tmp/pear/temp/phalcon -I/usr/local/include/php -I/usr/local/include/php/main -I/usr/local/include/php/TSRM -I/usr/local/include/php/Zend -I/usr/local
/include/php/ext -I/usr/local/include/php/ext/date/lib -DHAVE_CONFIG_H -g -O2 -D_GNU_SOURCE -DZEND_COMPILE_DL_EXT=1 -c /tmp/pear/temp/phalcon/phalcon.zep.c -MM
D -MF phalcon.dep -MT phalcon.lo  -fPIC -DPIC -o .libs/phalcon.o
/tmp/pear/temp/phalcon/phalcon.zep.c: In function 'zephir_function_exists':
/tmp/pear/temp/phalcon/phalcon.zep.c:4070:101: warning: comparison between pointer and integer
 4070 |  if (zend_hash_str_exists(CG(function_table), Z_STRVAL_P(function_name), Z_STRLEN_P(function_name)) != NULL) {
      |                                                                                                     ^~
/tmp/pear/temp/phalcon/phalcon.zep.c: In function 'zephir_function_exists_ex':
/tmp/pear/temp/phalcon/phalcon.zep.c:4079:76: warning: comparison between pointer and integer
 4079 |  if (zend_hash_str_exists(CG(function_table), function_name, function_len) != NULL) {
      |                                                                            ^~
/tmp/pear/temp/phalcon/phalcon.zep.c: In function 'zephir_preg_match':
/tmp/pear/temp/phalcon/phalcon.zep.c:8054:2: error: unknown type name 'pcre_cache_entry'
 8054 |  pcre_cache_entry *pce;
      |  ^~~~~~~~~~~~~~~~
/tmp/pear/temp/phalcon/phalcon.zep.c:8069:13: warning: implicit declaration of function 'pcre_get_compiled_regex_cache' [-Wimplicit-function-declaration]
 8069 |  if ((pce = pcre_get_compiled_regex_cache(Z_STR_P(regex))) == NULL) {
      |             ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/tmp/pear/temp/phalcon/phalcon.zep.c:8069:11: warning: assignment to 'int *' from 'int' makes pointer from integer without a cast [-Wint-conversion]
 8069 |  if ((pce = pcre_get_compiled_regex_cache(Z_STR_P(regex))) == NULL) {
      |           ^
/tmp/pear/temp/phalcon/phalcon.zep.c:8081:3: warning: implicit declaration of function 'php_pcre_match_impl' [-Wimplicit-function-declaration]
 8081 |   php_pcre_match_impl(pce, Z_STR_P(subject), return_value, &tmp_matches, global, 1, flags, offset);
      |   ^~~~~~~~~~~~~~~~~~~
/tmp/pear/temp/phalcon/phalcon.zep.c: In function 'zephir_require_ret':
/tmp/pear/temp/phalcon/phalcon.zep.c:9138:19: warning: passing argument 1 of 'zval_ptr_dtor' from incompatible pointer type [-Wincompatible-pointer-types]
 9138 |     zval_ptr_dtor(zend_string_path);
      |                   ^~~~~~~~~~~~~~~~
      |                   |
      |                   zend_string * {aka struct _zend_string *}
In file included from /usr/local/include/php/Zend/zend.h:36,
                 from /usr/local/include/php/main/php.h:31,
                 from /tmp/pear/temp/phalcon/phalcon.zep.c:46:
/usr/local/include/php/Zend/zend_variables.h:79:35: note: expected 'zval *' {aka 'struct _zval_struct *'} but argument is of type 'zend_string *' {aka 'struct
_zend_string *'}
   79 | ZEND_API void zval_ptr_dtor(zval *zval_ptr);
...
</code>
</pre>
</details>

**Details**
 - Phalcon version: 5.2.0
 - PHP Version: 8.2
 - Operating System: Windows w/ docker

**Additional context**

In `configure` command is detected `HAVE_BUNDLED_PCRE(ZEPHIR_USE_PHP_PCRE)` , but `ext/pcre/php_pcre.h` is not included in make, which may cause an error.
When running on PHP 8.2, it may be necessary to include it in `php_phalcon.h`.

### LINKED ISSUE #16295: [BUG]: build failure with old pcre2 version 10.23
Trying to build on RHEL/CentOS 7

```
/builddir/build/BUILD/php-phalcon5-5.2.0/NTS/phalcon.zep.c:8054:2: error: unknown type name 'pcre_cache_entry'
 8054 |  pcre_cache_entry *pce;
      |  ^~~~~~~~~~~~~~~~
/builddir/build/BUILD/php-phalcon5-5.2.0/NTS/phalcon.zep.c:8069:13: warning: implicit declaration of function 'pcre_get_compiled_regex_cache' [-Wimplicit-function-declaration]
 8069 |  if ((pce = pcre_get_compiled_regex_cache(Z_STR_P(regex))) == NULL) {
      |             ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~

```

This only affects PHP 8.2 (build OK with PHP 8.1)

### PR DESCRIPTION
**Pull request by Jeckerson**
Hello!

Type: bug fix

**In raising this pull request, I confirm the following:**

- [x] I have read and understood the [Contributing Guidelines](https://github.com/phalcon/cphalcon/blob/master/CONTRIBUTING.md)
- [x] I have checked that another pull request for this purpose does not exist
- [ ] I wrote some tests for this PR
- [ ] I have updated the relevant CHANGELOG
- [ ] I have created a PR for the [documentation](https://github.com/phalcon/docs) about this change

Small description of change:

Thanks

### CODE REVIEW

File: build/config/phalcon_c_header.php [hunk 1]
```diff
@@ -56,6 +56,10 @@
 #include <ext/spl/spl_iterators.h>
 #include <ext/spl/spl_array.h>
 
+#if PHP_VERSION_ID >= 80200
```
Comment (remicollet) [inline build/config/phalcon_c_header.php:59]: I don't think the condition is needed.
This is part of the PHP API, always there

The failure on 8.2 may be related to the include of this file in another header.

Comment (remicollet) [reply build/config/phalcon_c_header.php:59]: IMHO, better to use:

`#if defined ZEPHIR_USE_PHP_PCRE && ZEPHIR_USE_PHP_PCRE`

Comment (Jeckerson) [reply build/config/phalcon_c_header.php:59]: Agree

**Review by remicollet — APPROVED** [review]
Build with success using revelant part:
, https://git.remirepo.net/cgit/rpms/php/php-phalcon5.git/tree/phalcon-php82.patch

**Review by niden — APPROVED** [review]

**Comment by niden** [comment]
Thank you @remicollet

**Pull request merged**